include/solvers/multiphysics_interface.h

```diff
@@ -540,6 +540,46 @@ class MultiphysicsInterface
     return reynolds_stress_solutions;
   }
 
+  /**
+   * @brief Request the solid objects. Used when a heat boundary condition
+   * is applied on a solid, through Nitsche immersed boundary method.
+   */
+  std::vector<std::shared_ptr<SolidBase<dim, dim>>> *
+  get_solids()
+  {
+    return solids;
+  }
+
+  /**
+   * @brief Request the matrix of a given physics
+   *
+   * @param physics_id The physics of the solution being requested
+   */
+  TrilinosWrappers::SparseMatrix *
+  get_matrix(PhysicsID physics_id)
+  {
+    AssertThrow((std::find(active_physics.begin(),
+                           active_physics.end(),
+                           physics_id) != active_physics.end()),
+                ExcInternalError());
+    return physics_matrix[physics_id];
+  }
+
+  /**
+   * @brief Request the rhs of a given physics
+   *
+   * @param physics_id The physics of the solution being requested
+   */
+  TrilinosWrappers::MPI::Vector *
+  get_rhs(PhysicsID physics_id)
+  {
+    AssertThrow((std::find(active_physics.begin(),
+                           active_physics.end(),
+                           physics_id) != active_physics.end()),
+                ExcInternalError());
+    return physics_rhs[physics_id];
+  }
+
```

Comment:
Do you really need those? Why can't the heat transfer solver itself apply the restriction. Passing RHS and matrices around seems off...

Reply:
This has been removed since they are not used anymore